tests/system/src/test/java/test/util/Util.java

```diff
@@ -377,15 +374,13 @@ public static void startup(CountDownLatch startupLatch, Runnable r) {
     }
 
     /**
-     * This synchronous method first hides all the specified stages (ignoring any
-     * null Stages) in the platform thread, then calls {@link Platform.exit()}.
+     * This synchronous method first hides all the open {@code Window}s in the platform thread,
+     * then invokes {@link Platform.exit()}.
      */
-    public static void shutdown(Stage... stages) {
+    public static void shutdown() {
         runAndWait(() -> {
-            for (Stage s : stages) {
-                if (s != null) {
-                    s.hide();
-                }
+            for (Window w : new ArrayList<>(Window.getWindows())) {
+                w.hide();
```

Comment:
I think you can use `List.copyOf` instead of an `ArrayList` since it doesn't get modified (it's a one-time copy).
The iteration can also be
```java
List.copyOf(Window.getWindows()).forEach(Window::hide);
```
if you want.

Reply:
The only issue with this style is - it's hard to set breakpoints.
In this case it's not that important.
Thank you!

Reply:
It's a matter of style mostly, so I don't really mind it. The debug issue can be remedied by adding breakpoints inside Lambdas, which all IDEs support I think (Eclipse and IntelliJ for sure).

Reply:
"inside lambdas" - inside the referenced methods you mean?  yes, but if that method is frequently called it's not what I want.  There is no way to set a breakpoint at the moment of lambda invocation, I think, at least not in Eclipse.